### ISSUE TITLE
zsh: illegal hardware instruction" error on Mac OS, M1 chip

### ISSUE THREAD

**Issue by MarcoEng77** [sintel-dev/Orion#575]
* Orion version: 0.6.1
* Python version: 3.11
* Operating System: Mac OS (M1 chip)

### Description

Trying to creare an `Orion` onject, the Jupyter notebook hangs forever, without crashing or erroring.
Running the same commands in a script, I get the error `zsh: illegal hardware instruction`

### What I Did

I installed `orion-ml` using pip as recommended in a freshly minted conda environment, with Python 3.11.
I then tried a "smoke test:

`from orion.data import load_signal
from orion import Orion
train_data = load_signal('S-1-train')
orion = Orion()`

and the jupyter notebook hung forever (>10 minutes).
I tried using a python script and I got the error `zsh: illegal hardware instruction`

**Comment by sarahmish** [comment]
Hi @MarcoEng77, thank you for using Orion!

Unfortunately, I can't immediately see why the test you have done is not working. I tried it myself and everything runs as expected. I did alter the hyperparameters such that we can see the training epochs.
```
Python 3.11.10 (main, Oct  3 2024, 02:26:51) [Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from orion.data import load_signal
>>> from orion import Orion
>>> train_data = load_signal('S-1-train')
>>> orion = Orion(hyperparameters={'keras.Sequential.LSTMTimeSeriesRegressor#1': {'epochs': 5, 'verbose': True}})
>>> orion.fit(train_data)
/opt/anaconda3/envs/orion-py11/lib/python3.11/site-packages/sklearn/impute/_base.py:356: FutureWarning: The 'verbose' parameter was deprecated in version 1.1 and will be removed in 1.3. A warning will always be raised upon the removal of empty columns in the future version.
  warnings.warn(
WARNING:absl:At this time, the v2.11+ optimizer `tf.keras.optimizers.Adam` runs slowly on M1/M2 Macs, please use the legacy Keras optimizer instead, located at `tf.keras.optimizers.legacy.Adam`.
Epoch 1/5
33/33 [==============================] - 8s 194ms/step - loss: 0.4498 - mse: 0.4498 - val_loss: 0.4378 - val_mse: 0.4378
Epoch 2/5
33/33 [==============================] - 6s 182ms/step - loss: 0.4227 - mse: 0.4227 - val_loss: 0.4316 - val_mse: 0.4316
Epoch 3/5
33/33 [==============================] - 6s 189ms/step - loss: 0.4161 - mse: 0.4161 - val_loss: 0.4259 - val_mse: 0.4259
Epoch 4/5
33/33 [==============================] - 6s 192ms/step - loss: 0.4139 - mse: 0.4139 - val_loss: 0.4290 - val_mse: 0.4290
Epoch 5/5
33/33 [==============================] - 6s 193ms/step - loss: 0.4085 - mse: 0.4085 - val_loss: 0.4219 - val_mse: 0.4219
```

Can you share a google colab notebook showing the problem you are running into? You can also try running the following [notebook](https://colab.research.google.com/drive/1oYTqKe7GGIgvB7VffH0xllg77zUBpd7S?usp=drive_link).

**Comment by MarcoEng77** [comment]
@sarahmish thank you very much for looking into this.
I cannot use colab notebooks, but all was needed to reproduce my error was to run
```
from orion.data import load_signal
from orion import Orion
train_data = load_signal('S-1-train')
orion = Orion()
```
and I would get the error `zsh: illegal hardware instruction` (in the terminal, running the code as a script: the VS code jupyter notebook would simply run forever).

I have been spending quite some time tracking this down, I think it boils to the (sadly too well known!) difficulties in using Tensorflow on Mac Silicon chips (in my case M1 one). I am surprised your installation has been tested to work on MacOS M1, to the best of my knowledge the simple `pip install tensorflow` still does not work for Mac M1. 

For now, the best (sort of) workaround I got is the following 
I first install Tensorflow (2.14.1, as I see on your `setup.py` that Orion uses <2.15) using the procedure  recommended on Mac M1, that is, using Miniforge and installing `tensorflow-macos` and `tensorflow-metal`.
In this same environment, I run `pip install ml-orion`, the pip installer leaves the tensorflow package untouched, and I can run your basic examples.

Will continue to try to figure it out. I take it you have tested the installation  on MacOS M1,need to understand why it would not work in my case Thank you for your support and for this great library

**Comment by sarahmish** [comment]
Definitely, I expect so. I opened another issue https://github.com/sintel-dev/Orion/issues/578 to mark this as a request for future releases.

For the time being, here are the versions that are installed on my machine where I have an M2 chip
```
tensorflow==2.14.1
tensorflow-estimator==2.14.0
tensorflow-io-gcs-filesystem==0.37.1
tensorflow-macos==2.14.1
```
